### PR TITLE
Use tokens to authenticate SQL connections.

### PR DESCRIPTION
**Pull request by dimas-b**

### CODE REVIEW

File: sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java [hunk 1]
```diff
@@ -34,31 +36,33 @@
...
+          String username;
           String password = info.get("password");
-          if (username == null || password == null) {
+          if (password == null) {
             throw new IllegalArgumentException("Missing credentials in connection properties.");
           }
 
           try {
-            authenticator.createToken(username, password);
+            StoredCredentials credentials = authenticator.validateToken(password);
```
Comment (dougwettlaufer) [inline sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:58]: Don't we want this to support both? Similar to how cql module works?

Comment (dimas-b) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:58]: CQL validates passwords through SPNEGO, IIRC... but it does not look like it maps well to JDBC... there's no API to negotiate with the real client, AFAIK... We could simulate that on the server side, of course.

Comment (dimas-b) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:58]: Ah, CQL uses SASL, but still it does not map well to JDBC :)

Comment (dimas-b) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:58]: I'd prefer to defer actual password validation if tokens are ok short term. WDYT?

Comment (dougwettlaufer) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:58]: Sorry that should have been phrased better. I meant don't we want to support both traditional username/password auth _and_ token based auth?

Comment (dimas-b) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:58]: Another option is to add an new method to `AuthenticationService` to validate passwords without creating a token. This looks goable short term. Shall I open a PR to `master` for that?

Comment (dimas-b) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:58]: As discussed offline, we're going to defer password-validation for SQL connections until we have a core API in Stargate for password checks. Previously (before this PR) passwords were validated by _creating a token_, which looks like an overkill.

File: sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java [hunk 2]
```diff
@@ -34,31 +36,33 @@
...
     this.authenticator = authenticator;
   }
 
   private void newConnection(ConnectionHandle ch, Map<String, String> info) {
     connections.computeIfAbsent(
         ch.id,
         sid -> {
-          String username = info.get("user");
+          String username;
           String password = info.get("password");
```
Comment (mpenick) [inline sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:52]: Could the be called "token" as to differentiate it from user/pass authentication in the future?

Comment (mpenick) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:52]: Ah, nm. I see how it's being passed.

Comment (dimas-b) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:52]: The name "password" is kind of fixed by JDBC, IIRC. We could require "token" to be the user name for clarity, WDYT?

Comment (mpenick) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java:52]: This is what CQL does. It uses a configurable setting for the token user's name, but it defaults to "token".

Comment (dimas-b) [reply sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java]: added code to enforce the "token" user name in JDBC connections.

**Review by dougwettlaufer — APPROVED** [review]

**Pull request merged**